apps/web/src/components/Posts/SharePostModal.tsx

```diff
@@ -0,0 +1,252 @@
+        post: postById(id: $postId) {
+          ... on ErrPostNotFound {
+            __typename
+          }
+          ... on Post {
+            __typename
+            author @required(action: THROW) {
+              username
+              profileImage {
+                ... on TokenProfileImage {
+                  token {
+                    ...getPreviewImageUrlsInlineDangerouslyFragment
+                  }
+                }
+                ... on EnsProfileImage {
+                  __typename
+                  profileImage {
+                    __typename
+                    previewURLs {
+                      medium
+                    }
+                  }
+                }
+              }
+            }
+            caption
+            tokens {
+              ...getPreviewImageUrlsInlineDangerouslyFragment
+            }
+          }
+        }
+      }
+    `,
+    { postId: postId }
+  );
+
+  const { post } = queryResponse;
+  const postUrl = `${getBaseUrl()}/post/${postId}`;
+
+  const { pushToast } = useToastActions();
+
+  const handleCopyToClipboard = useCallback(async () => {
+    void navigator.clipboard.writeText(postUrl);
+    pushToast({ message: 'Copied link to clipboard', autoClose: true });
+  }, [postUrl, pushToast]);
+
+  const handleShareButtonClick = useCallback(
+    (baseComposePostUrl: string) => {
+      const message = `I just posted ${tokenName} on gallery ${postUrl}`;
+      const encodedMessage = encodeURIComponent(message);
+
+      window.open(`${baseComposePostUrl}?text=${encodedMessage}`, '_blank');
+    },
+    [tokenName, postUrl]
+  );
+
+  // stripped down version of the pfp retrieving logic in ProfilePicture.tsx
+  const profileImageUrl = useMemo(() => {
+    if (post?.__typename !== 'Post') {
+      return null;
+    }
+
+    const { token, profileImage } = post.author.profileImage ?? {};
+
+    if (profileImage?.previewURLs?.medium) {
+      return profileImage.previewURLs.medium;
+    }
+
+    if (token) {
+      const result = getPreviewImageUrlsInlineDangerously({
+        tokenRef: token,
+      });
+      if (result.type === 'valid') {
+        return result.urls.large;
+      }
+      return null;
+    }
+
+    return null;
+  }, [post]);
+
+  if (post?.__typename !== 'Post') {
+    return null;
+  }
+
+  const token = post.tokens?.[0];
+  if (!token) {
+    return null;
+  }
+
+  const result = getPreviewImageUrlsInlineDangerously({ tokenRef: token });
+  if (result.type !== 'valid') {
+    return null;
+  }
+
+  const shareButtonsDetails = [
+    {
+      icon: <FarcasterIcon fill="#FEFEFE" />,
+      title: 'WARPCAST',
+      baseComposePostUrl: 'https://warpcast.com/~/compose',
+    },
+    {
+      icon: <LensIcon fill="#FEFEFE" />,
+      title: 'LENS',
+      baseComposePostUrl: 'https://hey.xyz/',
+    },
+    {
+      icon: <TwitterIcon fill="#FEFEFE" />,
+      title: 'TWITTER',
+      baseComposePostUrl: 'https://twitter.com/intent/tweet',
+    },
+  ];
```

Comment:
since this is just a static array this should be instantiated outside of the component. will move it up!